service/gcs/core/gcs/gcs.go

```diff
@@ -359,11 +377,16 @@ func (c *gcsCore) ListProcesses(id string) ([]runtime.ContainerProcessState, err
 	c.containerCacheMutex.Lock()
 	defer c.containerCacheMutex.Unlock()
 
-	if _, ok := c.containerCache[id]; !ok {
+	containerEntry := c.getContainer(id)
+	if containerEntry == nil {
 		return nil, errors.WithStack(gcserr.NewContainerDoesNotExistError(id))
 	}
 
-	processes, err := c.Rtime.GetAllContainerProcesses(id)
+	if containerEntry.container == nil {
+		return nil, nil
```

Comment:
I don't think so. When you haven't created the first process, then the list of processes is empty.